### PR TITLE
doc: #CO-1740 add documentation

### PR DESCRIPTION
**Pull request by BastienBenard**
## Describe your changes
Add Java Doc API
Add first test of OpenAPI for confluence
https://confluence-test.support-ent.fr/display/MSP/Test+OpenAPI

## Issue ticket number and link
[CO-1740](https://jira.support-ent.fr/browse/CO-1740)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/java/fr/openent/supportpivot/enums/SourceEnum.java [hunk 1]
```diff
@@ -1,12 +0,0 @@
...
-public enum SourceEnum {
-    ENT("ENT"),
-    JIRA("JIRA"),
-    LDE("LDE");
-
-    private final String name;
-    public String toString(){return name;}
-
-    SourceEnum(String name) { this.name = name;   }
-}
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/enums/SourceEnum.java:12]: Inutilisé

File: src/main/java/fr/openent/supportpivot/model/endpoint/jira/JiraEndpoint.java [hunk 2]
```diff
@@ -245,7 +245,7 @@ private URI prepareSearchRequest(JiraSearch jiraSearch) {
                     this.getClass().getSimpleName()));
         }
         filter.addFields(creationFieldJira);
-        filter.maxResults(jiraSearch.getMaxResult());
+        filter.setMaxResults(jiraSearch.getMaxResult());
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/model/endpoint/jira/JiraEndpoint.java:248]: https://github.com/OPEN-ENT-NG/support-pivot/pull/39#discussion_r1064437190

File: src/main/java/fr/openent/supportpivot/services/impl/JiraServiceImpl.java [hunk 3]
```diff
@@ -107,16 +112,6 @@ public JiraServiceImpl() {
...
-    /**
-     * Get a modules name ENT-like, returns the module name PIVOT-like
-     * Returns original name by default
-     * @param moduleName ENT-like module name
-     * @return PIVOT-like module name encoded in UTF-8
-     */
-    private String moduleEntToPivot(String moduleName) {
-        return Supportpivot.applicationsMap.getOrDefault(moduleName, Field.NOTEXIST);
-    }
-
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/services/impl/JiraServiceImpl.java:119]: Réorganisation/déplacement de code

File: src/main/java/fr/openent/supportpivot/services/impl/JiraServiceImpl.java [hunk 4]
```diff
@@ -151,6 +146,39 @@ else if (pivotTicket.getIdExterne() != null) {
...
+                    LOGGER.error(String.format("[SupportPivot@%s::getFromJira] Error when calling URL : %s : %s %s ",
+                            this.getClass().getSimpleName(), urlGetTicketGeneralInfo, response.statusCode(), response.statusMessage()));
+                    LOGGER.error(response.body().toString());
+                    promise.fail("Error when gathering Jira ticket information");
+                }
+            }
+        });
+
+        return promise.future();
+    }
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/services/impl/JiraServiceImpl.java:174]: Réorganisation/déplacement de code

File: src/main/java/fr/openent/supportpivot/services/impl/MongoServiceImpl.java [hunk 5]
```diff
@@ -53,11 +53,10 @@ public Future<Void> saveTicket(String source, JsonObject jsonTicket) {
     }
 
     @Override
-    public Future<JsonObject> getMongoInfos(String mailTo) {
+    public Future<JsonObject> getMongoInfos(JsonObject request) {
         Promise<JsonObject> promise = Promise.promise();
         try {
-            JsonObject req = new JsonObject(java.net.URLDecoder.decode(mailTo, Field.ENCODE_UTF_8));
-            mongo.find(mongoCollection, req, MongoDbResult.validResultHandler(FutureHelper.handlerEitherPromise(promise)));
+            mongo.find(mongoCollection, request, MongoDbResult.validResultHandler(FutureHelper.handlerEitherPromise(promise)));
```
Comment (BastienBenard) [inline src/main/java/fr/openent/supportpivot/services/impl/MongoServiceImpl.java:59]: Ce n'est pas de le responsabliliter de MongoService de decoder l'URL

**Review by qpic-cgi — APPROVED** [review]

File: src/main/java/fr/openent/supportpivot/services/MongoService.java [hunk 6]
```diff
@@ -4,7 +4,20 @@
 import io.vertx.core.json.JsonObject;
 
 public interface MongoService {
+    /**
+     * Sauvegarde un ticket au format Json dans la base de donnée mongo
```
Comment (sjollois) [inline src/main/java/fr/openent/supportpivot/services/MongoService.java:8]: en anglais pour être homogène avec le reste de la doc

File: src/main/java/fr/openent/supportpivot/services/impl/MongoServiceImpl.java (hunk 5 above)
Comment (sjollois) [reply src/main/java/fr/openent/supportpivot/services/impl/MongoServiceImpl.java:59]: tu peux retirer le try catch du coup ?

**Review by sjollois — APPROVED** [review]

File: src/main/java/fr/openent/supportpivot/services/MongoService.java (hunk 6 above)
Comment (BastienBenard) [reply src/main/java/fr/openent/supportpivot/services/MongoService.java:8]: bien vu

**Pull request merged**